modules/ROOT/pages/correlation-id.adoc

```diff
@@ -0,0 +1,252 @@
+= Correlation ID
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: tracing, correlation, id, anypoint, studio
+
+When Mule creates a new event, it generates a Java Universally Unique Identifier (UUID) string identifier called a *correlation ID* before sending the event to the next processor in the flow. This ID enables you to correlate different log entries with a particular execution.
+
+Use the correlation ID to understand the entire history of an event that resulted in an issue or an unhandled error (all errors include the correlation ID of the event that triggered it). Also, knowing the correlation ID helps you perform a memory analysis of heap dumps.
+
+To obtain a correlation ID, Mule first checks for one in the source message (for example, a JMS message or an HTTP listener with the X-CORRELATION-ID header). If the source does not set a correlation ID, Mule generates one using the correlation ID generator.
+
+== Modify the Correlation ID Generator
+
+Although it is best to avoid making changes to the correlation ID generator, you might need to format the correlation ID for the events if:
+
+* Your company has its own standard or format for correlation IDs.
+* Your external system correlation IDs are a different length than those in Mule, causing an incompatibility that makes them untraceable.
+
+To change how Mule generates the correlation ID:
+
+. Add the `<configuration>` component to your application XML.
+. Set the `correlationIdGeneratorExpression` attribute to specify the expression that generates the correlation ID:
++
+[source,xml,linenums]
+----
+<configuration correlationIdGeneratorExpression="#[<custom_generator_expression>]"/>
+----
+
+[IMPORTANT]
+Ensure that the algorithm you use cannot generate the same correlation ID for two different events.
+
+=== Correlation ID Generator Configuration Example
+
+In the following example, Mule generates the correlation ID by calling the Java `uuid()` method, and then replaces the dashes (`-`) with dots (`.`):
+
+[source,xml,linenums]
+----
+<configuration correlationIdGeneratorExpression="#[uuid() replace /-/ with('.')]"/>
+
+<http:listener-config name="HTTP_Listener_config">
+  <http:listener-connection host="0.0.0.0" port="8081" />
+</http:listener-config>
+<flow name="example">
+  <http:listener config-ref="HTTP_Listener_config" path="/test"/>
+  <set-payload value="some"/>
+  <logger level="WARN" message="#[payload]"/>
+</flow>
+----
+
+After executing the flow, the logger shows the following:
+
+----
+WARN  2021-03-30 16:46:11,269 [[MuleRuntime].uber.06: [test-project-app].example.CPU_LITE @6d3b5ad] [processor: example/processors/1; event: b5a95a63.f190.4c91.880a.5b54512fa6b1] org.mule.runtime.core.internal.processor.LoggerMessageProcessor: some
```

Comment:
Maybe in this section you can mention that the correlation ID for this example is b5a95a63.f190.4c91.880a.5b54512fa6b1 because this is the first time in the doc that shows an example of how an ID looks like.
Users might need to scroll down to the section _Example of Identifying the Correlation ID in Logs_ to have a first glance of a correlation ID example.